### PR TITLE
ROX-11323 Add labels pool in fake workloads

### PR DESCRIPTION
**Pull request by lvalerom**
## Description

Follow up PR to #3341. Creates a pool of labels to ensure label matching between resources.

## Checklist
- [ ] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

Use a workload with `matchLabels: true` configured.

```
go run tools/local-sensor/main.go -record -with-fakeworkload=workload.yaml
```

Labels used by services and network policies should match labels used by deployments

### CODE REVIEW

**Review by Maddosaurus — APPROVED** [review]
Looks good to me!
Out of interest: Do we expect any impact on performance when using non-matched labels, or was this added to create a test environment that's closer to reality?

**Comment by lvalerom** [comment]
> Out of interest: Do we expect any impact on performance when using non-matched labels, or was this added to create a test environment that's closer to reality?

TL;DR: we added this to get an environment closer to reality

If we use non-matched labels is very unlikely that we could have matching resources, therefore there is no way to guarantee that certain alerts will get triggered. Before this PR we could get all resources to match with all resources. That was good in the sense that we could guarantee the alerts getting triggered but it was not very realistic, thus this change.

Regarding the performance, in a totally random environment everything is a little bit.. random 🙂 it depends on how isolated the resources are. For example, in the unlikely situation in which all resources are isolated from each other then the performance will be better because we don't need to iterate through a lot of resources to find any matching ones. I would say though that the performance should be similar in all cases.
